src/internal/operators/connect.ts

```diff
@@ -0,0 +1,44 @@
+/** @prettier */
+import { lift } from '../util/lift';
+import { Subscriber } from '../Subscriber';
+import { OperatorFunction, ObservableInput } from '../types';
+import { Observable } from '../Observable';
+import { Subject } from '../Subject';
+import { from } from '../observable/from';
+import { Subscription } from '../Subscription';
+
+function defaultConnector<T>() {
+  return new Subject<T>();
+}
+export function connect<T, R>({
+  connector = defaultConnector,
+  setup,
+}: {
+  connector?: () => Subject<T>;
+  setup: (shared: Observable<T>) => ObservableInput<R>;
+}): OperatorFunction<T, R> {
+  return (source: Observable<T>) => {
+    return lift(source, function (this: Subscriber<R>, source: Observable<T>) {
+      const subscriber = this;
+      let subject: Subject<T>;
+      try {
+        subject = connector();
+      } catch (err) {
+        subscriber.error(err);
+        return;
+      }
+
+      let result: Observable<R>;
+      try {
+        result = from(setup(subject.asObservable()));
+      } catch (err) {
+        subscriber.error(err);
+        return;
+      }
+
+      const subscription = result.subscribe(subscriber);
+      subscription.add(source.subscribe(subject));
+      return subscription;
+    });
```

Comment:
nitpick: This PR predates the smallification, so it should be updated to use `operate`. And the `source.subscribe(subject)` subscription should be added to the `subscriber`, etc.